tests/generate/conftest.py

```diff
@@ -1,24 +1,40 @@
 from importlib import reload
 
 import pytest
+import torch
 
 
-def pytest_collection_modifyitems(config, items):
-    """If mlxlm and Metal aren't available, skip mlxlm tests"""
+def is_mlx_available():
     try:
         import mlx.core as mx
         import mlx_lm  # noqa: F401
 
         assert mx.metal.is_available()
     except (ImportError, AssertionError):
-        skip_marker = pytest.mark.skip(
-            reason="Skipping test because mlx-lm or Metal are not available"
-        )
-        for item in items:
-            if "model_fixture" in item.fixturenames:
-                model_param = item.callspec.params.get("model_fixture", None)
-                if model_param == "model_mlxlm":
-                    item.add_marker(skip_marker)
+        return False
+    return True
+
+
+def is_cuda_available():
+    return torch.cuda.is_available()
+
+
+def pytest_collection_modifyitems(config, items):
+    """If mlxlm and Metal aren't available, skip mlxlm tests"""
+    skipped_models = []
+    if not is_mlx_available():
+        skipped_models.append("model_mlxlm")
+    if not is_cuda_available():
+        skipped_models.append("model_vllm")
+
+    skip_marker = pytest.mark.skip(
+        reason="Skipping test due to incompatible hardware / system."
+    )
+    for item in items:
+        if "model_fixture" in item.fixturenames:
+            model_param = item.callspec.params.get("model_fixture", None)
+            if model_param in skipped_models:
+                item.add_marker(skip_marker)
```

Comment:
These changes aren't directly related to the PR, but a minor change: I added vLLM integration tests to `test_generate.py`. This disables vLLM (and mlxlm) tests if the hardware is wrong.